doc/serial-timeout-troubleshoot.md

```diff
@@ -0,0 +1,80 @@
+# Troubleshooting "serial timeout" errors
+
+Several nRF Connect Desktop applications use serial ports to communicate with
+development kits and dongles. nRF Connect Desktop also uses serial ports to
+perform DFU (Device Firmware Update) when a devkit/dongle is running a 
+[DFU bootloader](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.0.0/sdk_app_serial_dfu_bootloader.html?cp=4_0_0_4_3_4).
+
+Serial ports might be [UARTs](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52840.ps/uart.html?cp=2_0_0_5_32)
+on the devkits connected to a RS-232 port on a computer,
+or they might be emulated through USB CDC ACM (also known as "Virtual COM port",
+"serial over USB" or other names). The later happens when using 
+[USB ACM CDC on the IMCU](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52/dita/nrf52/development/preview_dev_kit/vir_com_port.html?cp=2_1_5_4_1),
+or when using [USB functionality from the Nordic SDK](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.0.0/lib_usbd_class_cdc.html?cp=4_0_0_3_51_8_3)
+on SoCs that implement USB (e.g. the nRF52840).
+
+While serial ports are ubiquitous, serial port transport lacks any protocol
+metadata or capabilities for request-response higher-level protocols. Combined
+with the fact that modern desktop operating systems still allow for legacy
+serial ports, this can trigger timeouts.
+
+This document collects some known scenarios and workarounds for these timeout 
+errors.
+
+# Scenarios where these errors happen
+
+
+## DevKit/dongle not running bootloader
```

Comment:
This title might be misleading since the following sentences are talking about _performing a DFU_ which does happen when running the bootloader. To also cover the latest macOS issue I suggest to rephrase the title somehow like: 'Device not running bootloader or stuck in bootloader'.

Reply:
No, for the macOS issues I'd rather create a new section, not rephrase an existing one.

Or does the macOS issue get solved when manually triggering DFU mode?

Reply:
My suggestion is not about the macOS issue, however would be true for it. I think the title is misleading since some of scenarios you've described are already running the bootloader. The user sees a pulsing red LED, assumes that the device is in bootloader and skips the whole section.